src/tutorials/Mutable-File-System/05.vue

```diff
@@ -31,59 +31,66 @@ const validate = async (result, ipfs) => {
+    return { fail: 'Oops, you forgot to return a result. Did you accidentally delete `return directoryContents`?' }
   } else if (uploadedFiles = false) {
     // shouldn't happen because you can't hit submit without uploading files
-    return {'fail': 'Oops! You forgot to upload files to work with :('}
+    return { fail: 'Oops! You forgot to upload files to work with :(' }
   } else if (expected.length === 0) {
     // if no files are written to IPFS because they change the default code
-    return {'fail': 'Uh oh. Looks like no files made it into IPFS. Did you accidentally edit the default `write` code?'}
+    return { fail: 'Uh oh. Looks like no files made it into IPFS. Did you accidentally edit the default `write` code?' }
   } else if (!itemsAreFiles) {
     // if they forget the file name and just use a directory as the path
     // shouldn't happen unless they mess with default code
-    return {'fail': 'Uh oh. It looks like you created a folder instead of a file. Did you forget to include a filename in your path?'}
+    return { fail: 'Uh oh. It looks like you created a folder instead of a file. Did you forget to include a filename in your path?' }
   } else if (!rightFilesUploaded) {
-    return {'fail': 'Uh oh. Your files weren\'t added to IPFS correctly. Did you accidentally edit the default `write` code?'}
+    return { fail: 'Uh oh. Your files weren\'t added to IPFS correctly. Did you accidentally edit the default `write` code?' }
   } else if (result[0].hash.length === 0) {
-    return {'fail': 'Oops! Looks like you forgot to use the {long: true} option! Open your console to see what happened.'}
+    return {
+      fail: 'Oops! Looks like you forgot to use the { long: true } option!',
+      logDesc: 'When you forget to use the `{ long: true }` option, the `ls` method only returns filenames, not file types, sizes, or hashes (CIDs). Take a look at what happens without that option below, then try again.',
+      log: result
+    }
   } else if (directoryContentsMatch) {
-    return {'success': 'Success! Open your console to see the data returned by the `ls` method.'}
+    return {
+      success: 'Success! You did it!',
+      logDesc: 'Take a look at the complete data returned by the `ls` method:',
+      log: result
+    }
   } else {
-    return {'fail': 'Something we haven\'t anticipated is wrong. :('}
+    return { fail: "Something we haven't anticipated is wrong. :(" }
   }
 }
 
 const code = `const run = async (files) => {
-  // this code adds your uploaded files to your root directory in IPFS
-  await Promise.all(files.map(f => ipfs.files.write('/' + f.name, f, {create: true})))
-  // add your code to create a new directory here
-  let rootContents = await ipfs.files.ls('/', {long: true})
-  return rootContents
+  // this code adds your uploaded files to IPFS
+  await Promise.all(files.map(f => ipfs.files.write('/' + f.name, f, { create: true })))
```

Comment:
It's a bit odd to see this after we just did it with a loop. Why not just have the loop here again?